src/context.rs

```diff
@@ -1395,7 +1455,39 @@ impl BlockContext {
       *c = 0;
     }
   }
-  //TODO(anyone): Add reset_left_tx_context() here then call it in reset_left_contexts()
+
+  pub fn update_tx_size_context(
+    &mut self, bo: BlockOffset, bsize: BlockSize, tx_size: TxSize, skip: bool
+  ) {
+    let n4_w = bsize.width_mi();
+    let n4_h = bsize.height_mi();
+    let mut tx_w = tx_size.width() as u8;
+    let mut tx_h = tx_size.height() as u8;
+
+    let above_ctx =
+      &mut self.above_tx_context[bo.x..bo.x + n4_w as usize];
+    let left_ctx = &mut self.left_tx_context
+      [bo.y_in_sb()..bo.y_in_sb() + n4_h as usize];
+
+    if skip {
+      tx_w = n4_w as u8;
+      tx_h = n4_h as u8;
+    }
+
+    for i in 0..n4_w {
```

Comment:
``` rust 
for v in above_ctx[0..n4_w] {
  *v = tx_w;
}
```
maybe?

Reply:
Thanks again. (You meant " for v in above_ctx[0..n4_w].iter_mut()", btw )